L1Trigger/TrackFindingTMTT/interface/GlobalCacheTMTT.h

```diff
@@ -0,0 +1,56 @@
+#ifndef L1Trigger_TrackFindingTMTT_GlobalCacheTMTT_h
+#define L1Trigger_TrackFindingTMTT_GlobalCacheTMTT_h
+
+// Data shared across all streams by TMTT L1 tracking.
+//
+// Provides the python configuration parameters
+// & optional histogramming/debugging.
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Settings.h"
+#include "L1Trigger/TrackFindingTMTT/interface/HTrphi.h"
+#include "L1Trigger/TrackFindingTMTT/interface/StubWindowSuggest.h"
+#include "L1Trigger/TrackFindingTMTT/interface/TrackerModule.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Histos.h"
+
+#include <list>
+#include <memory>
+
+namespace tmtt {
+
+  class GlobalCacheTMTT {
+  public:
+    GlobalCacheTMTT(const edm::ParameterSet& iConfig)
+        : settings_(iConfig),              // Python configuration params
+          htRphiErrMon_({0., 0, 0, 0}),    // rphi HT error monitoring
+          stubWindowSuggest_(&settings_),  // Recommend FE stub window sizes.
+          hists_(&settings_)               // Histograms
+    {
+      hists_.book();
+    }
+
+    // Get functions
+    Settings& settings() const { return settings_; }
```

Comment:
Can this be
```suggestion
    const Settings& settings() const { return settings_; }
```
?

Based on
https://github.com/cms-sw/cmssw/blob/de9b0a8d8582aba8fa9c9f221c0e233879735d8e/L1Trigger/TrackFindingTMTT/plugins/TMTrackProducer.cc#L99
the answer appears to be "no", and that use is not thread safe.

Reply:
The magneticField_ variable inside Settings.h is of type std::atomic<float>. Therefore, I believe the use made of the Settings class in GlobalCacheTMTT is thread safe. No other variables inside Settings.h are modified during the run. The line in TMTrackProducer.cc you found loads a typical field value, which is used by the histogramming. If you prefer, we could  return type "const Settings&" from GlobalCacheTMTT, if we are allowed to make magneticField_ "mutable"?

Reply:
Thanks, I missed the fact that the `Settings::magneticField_` was atomic. And all other methods are `const`, so technically it is thread safe. If this is part of the planned rewrite, I'm fine with it as it is.

Reply:
I've changed GlobalCacheSettings::settings() to "const", but changed Settings::magnetic field to type "mutable std::atomic< float >". This makes clear that one is only allowed to change that one data member inside Settings. (It also facilitates the communication about the B-field with the TrackFindingTracklet package)